app/scripts/main.js

```diff
@@ -392,6 +402,84 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
       }
       return def.promise();
     },
+    getScatterData: function(_self, groupId) {
+      var def = new $.Deferred();
+      var self = this;
+      var data = self.getGroupNdx(groupId);
+
+      $.when(window.iviz.datamanager.getStudyToSampleToPatientdMap()).then(function(_studyToSampleToPatientMap) {
+        $.when(window.iviz.datamanager.getCaseLists()).then(function(_caseLists) {
+          var _sequencedCaseUIDs = [];
+          var _allCaseUIDs = [];
+          var _sequencedCaseUIdsMap = {};
+          
+          $.each(_caseLists, function(studyId, caseList) {
```

Comment:
You're right. Thanks! I removed them and made _sequencedCaseUIdsMap as a parameter sent to scatterPlotTemplate.js